brian2genn/device.py

```diff
@@ -1132,48 +1160,50 @@ def compile_source(self, debug, directory, use_GPU):
             else:
                 raise RuntimeError('Set the CUDA_PATH environment variable or '
                                    'the devices.genn.cuda_path preference.')
-        if prefs['codegen.cpp.extra_link_args']:
-            # declare the link flags as an environment variable so that GeNN's
-            # generateALL can pick it up
-            env['LINK_FLAGS'] = ' '.join(prefs['codegen.cpp.extra_link_args'])
+
```

Comment:
I dug a bit into old mails and apparently (back in 2017...), Thomas had the following issue on the cluster with the K40 when it tried to use the `gcc` that comes with conda:
```
 /usr/bin/ld: skipping incompatible /usr/lib/../lib/libm.so when searching for -lm
/usr/bin/ld: skipping incompatible /usr/lib/../lib/libc.so when searching for -lc
./generateALL_CPU_ONLY: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.14' not found (required by ./generateALL_CPU_ONLY) 
```
A solution was to use static linking by passing `-static-libstdc++` via `LINK_FLAGS`, not only for our `main` but also for the `generateAll` binary. This might not be necessary any more, but I'd still like to keep the mechanism. No idea of whether this is useful for Windows in any way, but I think your change removed it globally, i.e. also for Linux and OS X, right?

Reply:
sure - will revert that for non-windows. Is the ``-static-libstdc++`` thing always passed in or was that a manual hack on a particular machine?

Reply:
No that was just done for a single machine, I don't think we want to link statically by default. I don't quite remember the details, but there might have been other occasions where we used a link flag like that (for C++ standalone at least).